client/src/Client.js

```diff
@@ -0,0 +1,145 @@
+import axios from "axios";
+
+function checkStatus(response) {
+  if (response.status >= 200 && response.status < 300) {
+    return response;
+  } else {
+    const error = new Error(`${response.statusText}`);
+    error.status = response.statusText;
+    error.response = response;
+    throw error;
+  }
+}
+
+function parseData(response) {
+  return response.data;
+}
+
+export class ArticlesClient {
+  getArticle(data) {
+    return axios
+      .get("http://localhost:3001/api/articles", {
+        headers: {
+          Accept: "application/json",
+        },
+      })
+      .then(checkStatus)
+      .then(parseData)
+      .then((res) => res.find((article) => article.id === data.articleId));
+  }
+
+  getArticles() {
+    return axios
+      .get("/api/articles", {
+        headers: {
+          Accept: "application/json",
+        },
+      })
+      .then(checkStatus)
+      .then(parseData);
+  }
+
+  getUserArticles(data) {
```

Comment:
as a REST API it is better it change the URL to `/api/users/:id/articles`

Reply:
it would be great if you have a quick look on this https://restfulapi.net/rest-api-design-tutorial-with-example/

Reply:
Fixed